index.js

```diff
@@ -793,25 +797,28 @@ Command.prototype.normalize = function(args) {
 Command.prototype.parseArgs = function(args, unknown) {
   var name;
 
+  if (this.listeners('program-action').length && this.listeners('command:*').length) {
```

Comment:
I do not think need to error? My earlier comment about wondering what use case there was for both was about understanding existing patterns rather than about preventing both.

Reply:
Preventing both does make it easier though. We can’t really differentiate between the two. If someone has both and it’s not an error either they both get called which seems unexpected, or only one gets called and it won’t be immediately obvious which. 

I really do think it’d be good to error for this case.

Reply:
If someone has both then calling them both is backwards compatible, so that is my preferred behaviour.

I don't see why people would want to define both, but I don't see a significant benefit in blocking it. I don't recall seeing reported issues about people wanting different behaviours when defining both — either wanting just one called,  or being confused by calling both.

I do empathise with wanting to reduce the number of possible scenarios since we had trouble reasoning about it, but not at the cost of breaking code patterns we didn't think of.

Reply:
I updated this as suggested